src/modelFactory/FindFunctions.cpp

```diff
@@ -3,82 +3,90 @@
 //
 
 #include <iostream>
+#include <GlobalUtilities.h>
 #include "FindFunctions.h"
 #include "FatalError.h"
 #include "Logger/Logger.h"
-
-DESCAM::FindFunctions::FindFunctions() {}
-
-bool DESCAM::FindFunctions::setup(clang::CXXRecordDecl *recordDecl) {
-  this->clean();
-  return TraverseDecl(recordDecl);
-}
-
-void DESCAM::FindFunctions::clean() {
-  function_map_.clear();
-  function_return_type_map_.clear();
-  function_param_name_map_.clear();
-  function_param_type_map_.clear();
+#include "FindDataFlow.h"
+#include "CFGFactory.h"
+#include "FunctionFactory.h"
+
+using namespace DESCAM;
+
+DESCAM::FindFunctions::FindFunctions(IFindNewDatatype *find_new_datatype,
+                                     IFindDataFlowFactory *find_data_flow_factory) {
```

Comment:
Whenever possible, field assignments in CTORs should be done by using initializer lists e.g. Foo::Foo(in a, Bar* ptr) : a_(a), ptr_(ptr= {}